halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/services/v1/DeploymentEnvironmentService.java

```diff
@@ -56,14 +62,26 @@ public DeploymentEnvironment getDeploymentEnvironment(String deploymentName) {
 
   public void setDeploymentEnvironment(String deploymentName, DeploymentEnvironment newDeploymentEnvironment) {
     DeploymentConfiguration deploymentConfiguration = deploymentService.getDeploymentConfiguration(deploymentName);
+
+    String defaultLocation = lookupDefaultLocation(deploymentName, newDeploymentEnvironment.getAccountName());
+    String location = newDeploymentEnvironment.getLocation();
+    newDeploymentEnvironment.setLocation(StringUtils.isEmpty(location) ? defaultLocation : location);
```

Comment:
If we want to change the location whenever the account changes then we will have to do it on the CLI side of things, Do you have a better place in mind for the location change to take place or should we just make the change on the CLI side of things.